src/invert/laplacexz/impls/cyclic/laplacexz-cyclic.cxx

```diff
@@ -41,14 +41,18 @@ LaplaceXZcyclic::LaplaceXZcyclic(Mesh *m, Options *options, const CELL_LOC loc)
   k1d_2 = Array<dcomplex>((m->LocalNz) / 2 + 1);
 
   // Create a cyclic reduction object, operating on dcomplex values
-  cr = new CyclicReduce<dcomplex>(mesh->getXcomm(), nloc);
+  cr = new CyclicReduce<dcomplex>(localmesh->getXcomm(), nloc);
```

Comment:
Should this use `make_unique` as is done in `LaplaceXY`?

Reply:
I think so. At the moment `cr` causes a memory leak, since it is not deleted in the destructor...